### PR TITLE
Provide feedback when OpenXR can't start

### LINKED ISSUE #26213: Provide embedder feedback if openxr backend isn't set up right
Ideally we should pop up some kind of message dialog either on startup or when entering VR/AR fails. This requires some kind of coordination between the webxr backend, the DOM XR code, and the embedding API.

### PR DESCRIPTION
**Pull request by jdm**
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #26213
- [x] These changes do not require tests because it's testing an edge case in an environment we can't test on CI

### CODE REVIEW

**Comment by jdm** [comment]
Depends on https://github.com/servo/webxr/pull/164.

File: ports/libsimpleservo/api/src/lib.rs [hunk 1]
```diff
@@ -819,12 +819,27 @@ impl EmbedderMethods for ServoEmbedderCallbacks {
             }
         }
 
-        let discovery = webxr::openxr::OpenXrDiscovery::new(
-            Box::new(GlThread(executor)),
-            Box::new(ProviderRegistration(surface_providers)),
-            Box::new(ContextMenuCallback(embedder_proxy)),
-        );
-        registry.register(discovery);
+        if openxr::create_instance().is_ok() {
```
Comment (Manishearth) [inline ports/libsimpleservo/api/src/lib.rs:822]: Hmm, this would be nicer if we had the prompt when you attempt to enter immersive mode, IMO. Might need another error variant

**Review by Manishearth — APPROVED** [review]
r=me either way

File: ports/libsimpleservo/api/src/lib.rs (hunk 1 above)
Comment (jdm) [reply ports/libsimpleservo/api/src/lib.rs:822]: I'll file a followup for that, since it requires a bunch more plumbing.

**Comment by jdm** [comment]
@bors-servo retry
* #26315

**Comment by jdm** [comment]
@bors-servo retry
#25513
